File: 每日一题/5.1合并两个有序链表.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        preHeadNode = ListNode(0)
        iterPoint = preHeadNode
        while l1 != None and l2 != None:
            if l1.val < l2.val:
                iterPoint.next = l1
                l1 = l1.next
            else:
                iterPoint.next = l2
                l2 = l2.next
            iterPoint = iterPoint.next
        iterPoint.next = l1 if l1 else l2
        return preHeadNode.next

File: 每日一题/5.1合并两个有序链表.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode node(0); // dummy node
        ListNode *iter = &node; // iteration pointer
        while (l1 != nullptr && l2 != nullptr) {
            if (l1->val < l2->val) {
                iter->next = l1;
                l1 = l1->next;
            }
            else {
                iter->next = l2;
                l2 = l2->next;
            }
            iter = iter->next;
        }
        iter->next = l1 ? l1 : l2;
        return node.next;
    }
};